### PR TITLE
Add github actions to pull requests.

### PR DESCRIPTION
**Pull request by arne-z**
Add a github action running build, lint, and test jobs that runs automatically on pull requests.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+      - run: npm ci
+      - run: npm run ng build
```
Comment (ruegenbfr) [inline .github/workflows/build.yml:13]: Please use `npm run build:qa?` instead.

Comment (arne-z) [reply .github/workflows/build.yml:13]: Makes sense.

**Comment by arne-z** [comment]
Is this cleared to merge now @ruegenbfr?

**Review by ruegenbfr — CHANGES_REQUESTED** [review]
Please change node version from 18 to 16.

If you like, you can add a job for audit with the command

```
npm run audit -- --level=high --production 
```

**Pull request merged**